src/Core/BaseSettingsProgramOptions.h

```diff
@@ -11,8 +11,16 @@ template <typename T>
 void addProgramOptionAsMultitoken(T &cmd_settings, boost::program_options::options_description & options, std::string_view name, const typename T::SettingFieldRef & field)
 {
     auto on_program_option = boost::function1<void, const Strings &>([&cmd_settings, name](const Strings & values) { cmd_settings.set(name, values.back()); });
-    options.add(boost::shared_ptr<boost::program_options::option_description>(new boost::program_options::option_description(
-            name.data(), boost::program_options::value<Strings>()->multitoken()->composing()->notifier(on_program_option), field.getDescription())));
+    if (field.getTypeName() == "Bool")
+    {
```

Comment:
Given that l. 16/17 and l. 21/22 are almost the same, a brief comment could have been useful.